### ISSUE TITLE
DSMS Profile Order UI

### ISSUE THREAD

**Issue by ilominar** [51st-Vfw/JAFDTC#36]
Profile Order UI is a bit mysterious/confusing.

I assume this is useful for custom profiles so you can order a custom GBU-38 setup with the "default" GBU-38 setup? As custom profiles are not yet supported (and sounds like support is not very high on the TODO list), does it make sense to expose this UI in the first place?

If kept, would suggest the following changes,

1) Re-layout the page to have the check box on top of the ListView (easier to draw than describe, but something like row 0 CheckBox, row 1 ListView, row 2 TextBlock to support item 2).
2) Remove the hamburger drag affordances from the ListView rows, text in the UI can surface ability to drag (perhaps TextBlock below table with "Drag the profiles to re-order them"?).
3) "Order as shown" CheckBox should guard operation of the ListView: when unchecked, you should not be able to manipulate the list.
4) Unchecking the CheckBox should return the order of the items in the ListView to the default order
5) Naming should be consistent with the Munition Setting ListView where possible

Think these would also apply to the UI in the event it supports custom profiles (though I think I'd make some changes to what is in the list in that case, assuming I'm understanding the use case---I *think* the order of GBU-31 and GBU-38 is not relevant as these munitions won't ever use each other's profiles, so you could probably collapse all the base profiles into a "defaults" or "baseline" row and only show the custom profiles in the ListView).

**Comment by fizzle77** [comment]
I do want to keep this. It has a real and common purpose, even if the UI is a little weird.

The typical way to switch between weapons in the A-10 is with the DMS switch left/right on the stick. What this actually does is move among the weapon profiles that are set up. Slight changes in loadout will significantly change the order of these default profiles that exist when you start up. It's common to re-order them on the DSMS profiles page so that the order you're going through them with your HOTAS is consistent and predictable. That's the main purpose of this UI.

The text for a couple of these weapons is different than the other tab because that's how it is in the jet: the names on the main INV page are a little different from these profile names. We could make them match (except for Mavericks, which just get one profile called "MAVERICKS"), but I still think it's better to match the jet. The only other difference is the APKWS rockets, and it's very subtle: `M282L` vs. `M-282L`. We need to keep the subtly different names in the db file, so we can correlate with what we scrape from the display, but now that the configs are fully decoupled from the UI text, it's no big deal either way.

It's not very common to add new profiles, but if we wanted to add that capability, this would be the logical place. But we'd still want all the default profiles to appear here so they could be reordered in combination with whatever profiles the user added. There are a bunch more common cases I want to cover first in new systems (HMCS, TAD, TGP, maybe even CMDS) before I'd come back to this.

The checkbox probably isn't necessary. I thought it might be useful to have a way to temporarily leave this part of the DSMS config default, without pressing the button and resetting everything and without undoing any ordering changes you've made. But I can't think of a real scenario where you'd do that. If you don't care about the order, you just wouldn't mess with this list at all.

I kind of thought the drag handle affordance was nicer than a text description, but I don't feel strongly about it. The best "handles" I found among the Segoe glyphs maybe aren't obvious enough? Anyway I can certainly make that change.

To summarize, I propose:

1. Leave this tab in. It's useful.
2. Get rid of the checkbox altogether. It's probably not. :)
3. Get rid of the drag handles and add a text description at the bottom of the list.
4. Leave the small description difference for APKWS (and the big one of Mavericks) there, matching the jet. I'm almost indifferent, but 51/49 prefer preserving the jet's idiosyncrasies here.

**Comment by ilominar** [comment]
Catching up on this since the PR has already been merged... :)

That works.

On the handles, generally I've seen them hidden except in a "reorder" mode (that is typical on iOS; I was unable to find them in WinUI desktop stuff, though I didn't look very hard). I was coming at it from that perspective. Anyway, point is moot.
